camply/notifications/discord.py

```diff
@@ -0,0 +1,113 @@
+"""
+Push Notifications via Discord
+"""
+
+import logging
+from typing import List
+
+import requests
+
+from camply.config import DiscordConfig
+from camply.containers import AvailableCampsite
+from camply.notifications.base_notifications import BaseNotifications
+
+logger = logging.getLogger(__name__)
+
+
+class DiscordNotifications(BaseNotifications):
+    """
+    Push Notifications via Discord
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.session.headers.update({"Content-Type": "application/json"})
+        if any([DiscordConfig.DISCORD_WEBHOOK is None, DiscordConfig.DISCORD_WEBHOOK == ""]):
```

Comment:
There are a couple lines like this that haven't been formatted by black / ruff - that's why we're seeing linting errors.

[pre-commit](https://juftin.com/camply/contributing/#pre-commit) resolves this with [hatch](https://juftin.com/camply/contributing/#scripts) - so either of these commands will format the codebase for you:


```
hatch run format
```

```
pre-commit run --all-files
```

----
----

This part is kind of confusing. CI/CD runs both linting and type checking on the codebase - but only linting errors raise an error. type-checking will still flag issues on the PR but won't actually reject it.